File: src/impl/RunConfImporter.java
package impl;

import com.intellij.execution.RunnerAndConfigurationSettings;
import com.intellij.openapi.components.ServiceManager;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface RunConfImporter {
	static RunConfImporter getInstance(@NotNull Project project) {
		return ServiceManager.getService(project, RunConfImporter.class);
	}


	void importConfigs(List<RunnerAndConfigurationSettings> settingsList);

	void importConfigs(String importPath);

	List<RunnerAndConfigurationSettings> retrieveNewConfigs(String importPath);
}

File: src/impl/RunConfImporter.kt
package impl

import com.intellij.execution.RunnerAndConfigurationSettings
import com.intellij.openapi.components.ServiceManager
import com.intellij.openapi.project.Project

interface RunConfImporter {

    fun importConfigs(settingsList: List<RunnerAndConfigurationSettings>)

    fun importConfigs(importPath: String)

    fun retrieveNewConfigs(importPath: String): List<RunnerAndConfigurationSettings>

    companion object {
        fun getInstance(project: Project): RunConfImporter {
            return ServiceManager.getService(project, RunConfImporter::class.java)
        }
    }
}
